frontend/src/stores/roles/state.ts

```diff
@@ -1,6 +1,11 @@
+import { ref, type Ref } from 'vue'
 import type { Role } from '@/domain/Role'
-import { type Ref, ref } from 'vue'
+import type { Level } from '@/domain/Level'
 
-const roles: Ref<Role[]> = ref([])
+export interface RoleWithLevels extends Role {
```

Comment:
You don't need to do this, you already have a levels parameter inside Role object

![image](https://github.com/Saci-Career/saci-admin/assets/16262664/58165557-599b-446c-a5a0-8c081c5a93de)